src/celmath/randutils.cpp

```diff
@@ -87,8 +75,36 @@ constexpr inline float smooth(float t)
     return t * t * (3.0f - 2.0f * t);
 }
 
+constexpr inline std::uint32_t rot(std::uint32_t x, int k)
+{
+    return (x << k) | (x >> (32 - k));
+}
+
 } // end of anonymous namespace
 
+
+Jsf32::Jsf32(std::uint32_t seed) :
+    a(UINT32_C(0xf1ea5eed)), b(seed), c(seed), d(seed)
+{
+    for (int i = 0; i < 20; ++i)
+    {
+        (void)this->operator()();
```

Comment:
I think it's better to add a private function and call it here and in operator, cause this like looks cryptic